### DISCUSSION TITLE
So I got an hint to look at the best-practice doc page, but didn't found my answer there.

### DISCUSSION THREAD

**Discussion by benallard** [jobrunr/jobrunr#114] [Q&A] [answered]
The hint was about the serialisation taking too much time. The best-practice page actually advice us to pass id instead of full-blown entities for that purpose, but I didn't do that.

The log:

```
You are using a custom method (pkg.to.deriveddata.DerivedDataEventManager.getDerivedDataDao}(})) while enqueueing that takes a lot of time. See https://www.jobrunr.io/documentation/best-practices/ on how to use JobRunr effectively.
```

Here is my call to jobRunR:

``` java
BackgroungJob.enqueue(() -> getDerivedDataDao().callToMethod(bean.getProperty1(), bean.getProperty2(), bean.getProperty3()))
```

`getDerivedDataDao()` is a getter in the `DerivedDataEventManager` class getting an Injected bean (the setter is @Inject annotated).

What am I doing wrong?

Thanks.

**Comment by rdehuyss** [comment, marked as answer]
Hmm, that's strange.

Is it possible that the getDerivedDataDao() is lazy loaded? The warning is only printed if it takes more than 1 millisecond to complete (which is not that long, if I think about it). I can increase the time to 5 milliseconds.

You may be better off with using the generic method - that way, you don't even need to inject the DerivedDataDao.

So, the below should also work:

`BackgroungJob.<DerivedDataDao>enqueue((x -> x.callToMethod(bean.getProperty1(), bean.getProperty2(), bean.getProperty3()))`
